### PR TITLE
Port parallel planning reviews from MoveIt1

### PR DESCRIPTION
**Pull request by sjahr**
### Description

This PR ports addressed reviews from @rhaschke to MoveIt2. Most importantly, `bool generatePlan(...)` becomes const again :+1: 

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/plan_solutions.hpp [hunk 1]
```diff
@@ -39,6 +39,7 @@
 
 #include <algorithm>
 #include <moveit/planning_interface/planning_response.h>
+#include <mutex>
```
Comment (rhaschke) [inline moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/plan_solutions.hpp:42]: Why is that required now (and not before)?

**Review by rhaschke — COMMENTED** [review]
Only two nitpicks:

File: moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/plan_solutions.hpp (hunk 1 above)
Comment (sjahr) [reply moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/plan_solutions.hpp:42]: It build before, probably because we're lucky and the required header was included from somewhere else but I think it is nicer to ensure that everything is included.

**Review by rhaschke — APPROVED** [review]
